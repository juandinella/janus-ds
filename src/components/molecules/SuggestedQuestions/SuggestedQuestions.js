import isPropValid from '@emotion/is-prop-valid'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import styles from './SuggestedQuestions.styles'
import SuggestedQuestion from '../../atoms/SuggestedQuestion'
import Container from '../../layout/Container'

/**
 * SuggestedQuestions
 */

const shouldForwardProp = (prop) => isPropValid(prop)
const StyledSuggestedQuestions = styled.div.withConfig({ shouldForwardProp })`
  ${styles}
`

const SuggestedQuestions = ({ id, questions }) => {
  return (
    <StyledSuggestedQuestions id={id} data-testid={id}>
      <Container
        flex
        gap="xxs"
        overflow="auto"
        direction="column"
        justifyContent="flex-end"
        alignItems="flex-end"
        whiteSpace="nowrap"
        space="xs, 0"
      >
        {questions.map((question) => (
          <SuggestedQuestion key={question.id} id={question.id}>
            {question.children}
          </SuggestedQuestion>
        ))}
      </Container>
    </StyledSuggestedQuestions>
  )
}

SuggestedQuestions.displayName = 'SuggestedQuestions'

SuggestedQuestions.propTypes = {
  id: PropTypes.string,
  questions: PropTypes.node,
}

export default SuggestedQuestions
