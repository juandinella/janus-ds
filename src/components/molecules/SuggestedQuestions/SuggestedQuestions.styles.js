import { css } from 'styled-components'

export default css`
  display: flex;
  padding: 0 var(--spacing-xs);

  > div {
    &::-webkit-scrollbar {
      height: 4px;
      width: 4px;
      background: var(--color-neutral-300);
    }
    &::-webkit-scrollbar-thumb:horizontal {
      background: var(--color-neutral-700);
      border-radius: 6px;
    }
  }
`
