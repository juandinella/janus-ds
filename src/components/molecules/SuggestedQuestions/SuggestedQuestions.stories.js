import SuggestedQuestions from './SuggestedQuestions'

import { getTemplate } from '../../../helpers/storybook'

const Template = getTemplate(SuggestedQuestions)

let questionList = [
  {
    children: 'Question 1',
  },
  {
    children: 'Question number 2',
  },
  {
    children: 'Question number 3',
  },
  {
    children: 'Question number 4',
  },
  {
    children: 'Question number 5',
  },
  {
    children: 'Question number 6',
  },
  {
    children: 'Question number 7',
  },
  {
    children: 'Question number 8',
  },
  {
    children: 'Question number 9',
  },
  {
    children: 'Question number 10',
  },
]

export default {
  title: 'Molecules/SuggestedQuestions',
  component: SuggestedQuestions,
  args: {
    questions: [...questionList],
  },
  argTypes: {},
}

export const Default = Template.bind({})
