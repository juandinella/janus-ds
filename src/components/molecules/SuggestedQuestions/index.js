import SuggestedQuestions from './SuggestedQuestions'

export default SuggestedQuestions
