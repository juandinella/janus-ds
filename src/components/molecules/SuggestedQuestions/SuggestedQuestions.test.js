import initStoryshots from '@storybook/addon-storyshots'
import { render, screen, cleanup } from '@testing-library/react'
import React from 'react'
import SuggestedQuestions from './SuggestedQuestions'

initStoryshots({
  storyKindRegex: /^Molecules\/SuggestedQuestions$/,
})

let questionList = [
  {
    text: 'Question number 1',
  },
  {
    text: 'Question number 2',
  },
]
const componentProps = {
  id: 'unique_id',
  questions: [...questionList],
}

afterEach(cleanup)

describe('<SuggestedQuestions />', () => {
  test('should render component correctly', () => {
    render(<SuggestedQuestions {...componentProps} />)
    const comp = screen.getByTestId(componentProps.id)
    expect(comp).toBeInTheDocument()
  })
})
