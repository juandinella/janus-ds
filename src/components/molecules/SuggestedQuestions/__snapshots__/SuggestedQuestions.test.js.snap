// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/SuggestedQuestions Default 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <div
    className="sc-1tbsf4n-0 hpmbQe"
  >
    <div
      className="sc-1kqcgfr-0 hSdNZc"
      direction="column"
      overflow="auto"
    >
      <button
        className="sc-18ktvrn-0 hxGPnv"
        onClick={[Function]}
      >
        <span>
          Question 1
        </span>
      </button>
      <button
        className="sc-18ktvrn-0 hxGPnv"
        onClick={[Function]}
      >
        <span>
          Question number 2
        </span>
      </button>
      <button
        className="sc-18ktvrn-0 hxGPnv"
        onClick={[Function]}
      >
        <span>
          Question number 3
        </span>
      </button>
      <button
        className="sc-18ktvrn-0 hxGPnv"
        onClick={[Function]}
      >
        <span>
          Question number 4
        </span>
      </button>
      <button
        className="sc-18ktvrn-0 hxGPnv"
        onClick={[Function]}
      >
        <span>
          Question number 5
        </span>
      </button>
      <button
        className="sc-18ktvrn-0 hxGPnv"
        onClick={[Function]}
      >
        <span>
          Question number 6
        </span>
      </button>
      <button
        className="sc-18ktvrn-0 hxGPnv"
        onClick={[Function]}
      >
        <span>
          Question number 7
        </span>
      </button>
      <button
        className="sc-18ktvrn-0 hxGPnv"
        onClick={[Function]}
      >
        <span>
          Question number 8
        </span>
      </button>
      <button
        className="sc-18ktvrn-0 hxGPnv"
        onClick={[Function]}
      >
        <span>
          Question number 9
        </span>
      </button>
      <button
        className="sc-18ktvrn-0 hxGPnv"
        onClick={[Function]}
      >
        <span>
          Question number 10
        </span>
      </button>
    </div>
  </div>
</div>
`;
